plugins/behaviour/versionable/versionable.php

```diff
@@ -97,35 +66,33 @@ public function onTableAfterStore(CmsEvent\Table\AfterStoreEvent $event)
 			return;
 		}
 
-		if (!is_object($table) || !($table instanceof TableInterface))
+		if (!(is_object($table) && $table instanceof VersionableTableInterface))
 		{
 			return;
 		}
 
-		// Parse the type alias
-		$typeAlias = $this->parseTypeAlias($table);
-
-		// If the table doesn't support UCM we can't use the Versionable behaviour
-		if (is_null($typeAlias))
-		{
-			return;
-		}
+		// Get the Tags helper and assign the parsed alias
+		$typeAlias = $table->getTypeAlias();
+		$aliasParts = explode('.', $typeAlias);
 
-		// If the table doesn't have a content history helper we can't proceed
-		if (!property_exists($table, 'contenthistoryHelper'))
+		if (!(isset($aliasParts[0]) && ComponentHelper::getParams($aliasParts[0])->get('save_history', 0)))
```

Comment:
`$aliasParts[0]` is never `null`, but it can be empty string. The check for `save_history` should be reversed?

Reply:
I expect a developer to make sure that an object that implements the interface that the method returns the right data... Or is that asking to much?

save_history in 3.x is off by default. Should we enable this by default?

Reply:
I'd add the sanity check just to be safe. I'd never trust anyone not to do something stupid :D

Reply:
I also wouldn't enable by default.

Reply:
As long as there is also a sensible maximum versions then why not enable by default?

Reply:
The logic should be reversed like this:

```suggestion
		if ($aliasParts[0] === '' || !ComponentHelper::getParams($aliasParts[0])->get('save_history', 0)))
```